### PR TITLE
nicer naming system for runs

### PR DESCRIPTION
**Pull request by Ben-Epstein**
the core of this pr is the new `create_run_name` function + tests for it. 

User feedback suggested a nicer naming system. Instead of random names that are hard to parse, create run names that are today's date followed by the run_number that was created today.

tested with pytest, as well as:
With a new project
image

And an existing project
image

### CODE REVIEW

**Review by anthonycorletti — APPROVED** [review]
:shipit: 

on another note, this makes me think that searching by project name in the search pane might be useful.

**Comment by franz101** [comment]
b-e-a-t-i-f-u-l

File: dataquality/core/init.py [hunk 1]
```diff
@@ -113,6 +114,36 @@ def _set_labels_for_existing_run(project_name: str, run_name: str) -> None:
...
+    for run in runs:
+        # If it's not an auto-created run name, skip
+        if today not in run["name"]:
+            continue
+        splitter = f"{today}_"
+        run_num = run["name"].split(splitter)[-1]
+        if run_num.isdigit() and int(run_num) > max_run_today:
+            max_run_today = int(run_num)
+    run_num = max_run_today + 1
+    return f"{today}_{run_num}"
```
Comment (setu4993) [inline dataquality/core/init.py:144]: Nit: It might be nice to differentiate between user created names and auto-generated names. Maybe a simple prefix like `auto-2023...`.

Given we use dates, I can totally imagine users (like me!) who prefer to organize it by dates themselves.

File: dataquality/utils/name.py [hunk 2]
```diff
@@ -1612,9 +1612,9 @@ def validate_name(name: Optional[str], assign_random: bool = False) -> str:
 
 def random_name() -> str:
     result = [
-        str(uuid4())[:5],
         random.choice(ADJECTIVES),
         random.choice(COLORS),
         random.choice(ANIMALS),
+        str(uuid4())[:5],
```
Comment (setu4993) [inline dataquality/utils/name.py:1618]: Why did we move this to the end?

File: tests/core/test_init.py [hunk 3]
```diff
@@ -36,7 +39,9 @@
 @patch.object(ApiClient, "create_run")
 @patch("dataquality.core.init._check_dq_version")
 @patch.object(dataquality.core.init.ApiClient, "valid_current_user", return_value=True)
+@patch("dataquality.core.init.create_run_name", return_value="foo")
```
Comment (setu4993) [inline tests/core/test_init.py:42]: Nit: Given we're repeating this a few times in here, would definitely be nicer as a fixture.

File: dataquality/core/init.py [hunk 4]
```diff
@@ -113,6 +114,36 @@ def _set_labels_for_existing_run(project_name: str, run_name: str) -> None:
...
+    If a run_name is not passed into `init`, we create a run_name base with today's
+    date, and increment the digit at the end based on how many runs were created in
+    this project with this scheme.
+
+    ie:
+        2023-05-15-1
+        2023-05-15-2
+        2023-05-15-3
+        ...
+        2023-05-15-n
```
Comment (setu4993) [inline dataquality/core/init.py:129]: Looks like we are using `-` as the delimiter. In which case, the docstring is incorrect...
```suggestion
    ie:
        2023-05-15_1
        2023-05-15_2
        2023-05-15_3
        ...
        2023-05-15_n
```

**Review by setu4993 — APPROVED** [review]
Looks good. Left a few notes.

File: dataquality/core/init.py [hunk 5]
```diff
@@ -113,6 +114,36 @@ def _set_labels_for_existing_run(project_name: str, run_name: str) -> None:
...
+    """
```
[the remark this answers is not in the document]
Comment (Ben-Epstein) [reply dataquality/core/init.py:130]: Yea random had a few issues, and the only true feedback from customers has been they prefer this

File: dataquality/core/init.py [hunk 6]
```diff
@@ -113,6 +114,36 @@ def _set_labels_for_existing_run(project_name: str, run_name: str) -> None:
...
+    pid = api_client.get_project_by_name(project_name)["id"]
+    runs = api_client.get_project_runs(pid)
+    today = datetime.today().strftime("%Y-%m-%d")
+    max_run_today = 0
+    for run in runs:
+        # If it's not an auto-created run name, skip
+        if today not in run["name"]:
+            continue
+        splitter = f"{today}_"
```
[the remark this answers is not in the document]
Comment (setu4993) [reply dataquality/core/init.py:139]: How do you feel about `auto-YYYYMMDD-i`?

File: dataquality/utils/name.py (hunk 2 above)
Comment (Ben-Epstein) [reply dataquality/utils/name.py:1618]: Just arbitrary, I thought the uuid in the beginning looked pretty bad. And since we made that change just a few days ago, I don't think anyone's experienced it yet

File: dataquality/core/init.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (Ben-Epstein) [reply dataquality/core/init.py:139]: We can add auto, but it'll increase the complexity of the parsing. I'd be surprised if users date things exactly the same way as us?

File: tests/core/test_init.py (hunk 3 above)
Comment (Ben-Epstein) [reply tests/core/test_init.py:42]: Yea good point

File: dataquality/core/init.py (hunk 4 above)
Comment (Ben-Epstein) [reply dataquality/core/init.py:129]: Nice catch. I changed it halfway through, missed the docstring

File: dataquality/core/init.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (setu4993) [reply dataquality/core/init.py:139]: FWIW, `YYYY-MM-DD` is a pretty standard structure and the default for `str(date.today())`.
